src/omotes_orchestrator/timeout_job_manager.py

```diff
@@ -0,0 +1,126 @@
+# To avoid cyclic import error
+from __future__ import annotations
+from typing import TYPE_CHECKING, Optional
+if TYPE_CHECKING:
+    from omotes_orchestrator.main import Orchestrator
+
+import logging
+from datetime import datetime, timedelta
+import threading
+from omotes_orchestrator.postgres_interface import PostgresInterface
+from omotes_orchestrator.db_models.job import JobDB, JobStatus
+from omotes_sdk_protocol.job_pb2 import JobCancel
+from omotes_orchestrator.config import TimeoutJobManagerConfig
+
+LOGGER = logging.getLogger("omotes_orchestrator")
+
+
+class TimeoutJobManager:
+    """Periodically checks and cancels if any job is timed out."""
+
+    postgresql_if: PostgresInterface
+    """Interface to PostgreSQL."""
+    orchestrator: Optional[Orchestrator]
+    """Orchestrator application. Only when the orchestrator is linked with
+    the TimeoutJobManager instance, can the timeout_jobs_handler() method function properly."""
+    config: TimeoutJobManagerConfig
+    """TimeoutJobManager configuration."""
+    _init_time: datetime
+    """Instantiated datetime, is typically instantiated when the orchestrator is started."""
+    _start_buffer_sec: int
```

Comment:
As an alternative instead of using an annotated field name (e.g. _sec for seconds), you can also use datetime.timedelta to capture a duration of time. No need to change it as this works perfectly as well, but I at least wanted to note the option.

Reply:
Thanks for mentioning it! Will not change this implementation for now, but keep it in mind as an alternative.